.github/workflows/product-catalog-ci.yaml

```
@@ -0,0 +1,78 @@
+on:
+    pull_request:
+        branches:
+        - main
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        steps:
+        - name: checkout code
+          uses: actions/checkout@v4
+        - name: setup Go 1.22
+          uses: actions/setup-go@v2
```

actionlint (1.7.7)

the runner of "actions/setup-go@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)